core/src/main/java/org/elasticsearch/index/store/FsDirectoryService.java

```diff
@@ -127,6 +130,7 @@ protected Directory newFSDirectory(Path location, LockFactory lockFactory) throw
         throw new IllegalArgumentException("No directory found for type [" + storeType + "]");
     }
 
+    @Deprecated
```

Comment:
I don't think that we need to mark a private method as deprecated. The advantage of leaving this annotation off is that then you can move the two `IndexModule.Type.DEFAULT.match(storeType)` checks in `FsDirectoryService#newFSDirectory` to a separate private `isDefault` method (the JVM will inline it anyway) and thus move the suppression on `FsDirectoryService#newFSDirectory` to just that method which reduces the scope of the suppression.